src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -4926,8 +4926,13 @@ static QVariant executeGeomOverlay( const QVariantList &values, const QgsExpress
     request.setFilterExpression( filterString ); //filter cached features
   }
 
-  int limit = QgsExpressionUtils::getIntValue( values.at( 3 ), parent ); //in expressions overlay functions throw the exception: Eval Error: Cannot convert '' to int
-  request.setLimit( limit );
+  node = QgsExpressionUtils::getNode( values.at( 3 ), parent ); //in expressions overlay functions throw the exception: Eval Error: Cannot convert '' to int
```

Comment:
It seems that further optional parameters have to be interpreted as an expression and then converted to int. The result is a bit tricky, but goes.

Reply:
I don't know if it is the proper way to do this....